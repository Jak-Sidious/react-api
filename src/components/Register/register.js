// Component that deals with user registration
import React from 'react';
import Notifications, { notify } from 'react-notify-toast';
import { Form, Button } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import axiosInstance from '../commonComponents/AxiosInstance';

const REGISTRATION_URL = 'users/register'; // url for the registration of new users
class Register extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      username: '',
      email: '',
      password: ''
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleRegister = this.handleRegister.bind(this);
  }

  // function to handle changes made to the form
  handleChange(event) {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  // Function to handle the registration of new users
  handleRegister(event) {
    event.preventDefault();

    const newUser = {
      username: this.state.username,
      email: this.state.email,
      password: this.state.password
    };
    console.log(newUser);
    axiosInstance
      .post(`${REGISTRATION_URL}`, newUser)
      .then(response => {
        if (response.status === 201) {
          notify.show('User succesfully Registered');
          this.props.history.push('/login');
        }
      })
      .catch(error => {
        console.log(error.response);
        // if (
        //   error.response.data.message ===
        //   `Username ${this.state.username} already exists`
        // ) {
        //   notify.show(`Username ${this.state.username} already exists`);
        // } else if (
        //   error.response.data.message ===
        //   'Password must be between 6 and 25 alphanumeric characters.'
        // ) {
        //   notify.show(
        //     'Password must be between 6 and 25 alphanumeric characters.'
        //   );
        // } else if (
        //   error.response.data.message ===
        //   'Username is invalid it should contain alphanumeric charcaters
        // followed by an underscore of not more than 25 characters'
        // ) {
        //   notify.show('Username cannot start with a number');
        // }
      });
  }

  render() {
    return (
      <div className="regBackground">
        <Notifications />
        <div className="wrapper">
          <Form className="form-signin" onSubmit={this.handleRegister}>
            <h2 className="form-signin-heading">Registration form</h2>

            <Form.Input
              className="nameField"
              autoFocus
              name="username"
              type="username"
              placeholder="Enter your username"
              onChange={this.handleChange}
            />
            <Form.Input
              className="emailField"
              autoFocus
              name="email"
              type="email"
              placeholder="Enter email"
              onChange={this.handleChange}
            />
            <Form.Input
              className="passwordField"
              autoFocus
              name="password"
              type="password"
              placeholder="Enter password"
              onChange={this.handleChange}
            />
            <Button
              className="left floated"
              color="blue"
              value="submit"
              type="submit"
              size="large"
            >
              Register
            </Button>

            <Button
              className="right floated"
              action="/login"
              color="green"
              size="large"
              as={Link}
              to="/login"
            >
              Proceed to Login
            </Button>
            <br />
          </Form>
        </div>
      </div>
    );
  }
}

export default Register;
