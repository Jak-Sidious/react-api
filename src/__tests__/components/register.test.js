import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// import ReactDom from 'react-dom';
import { notify } from 'react-notify-toast';
import { MemoryRouter } from 'react-router-dom';

import Register from '../../components/Register/register';

Enzyme.configure({ adapter: new Adapter() });

describe('<Register/>', () => {
  const props = {
    post: jest.fn(() => Promise.resolve('register')),
    onChange: jest.fn(),
    handleChange: jest.fn(),
    preventDefault: jest.fn(),
    history: { push: jest.fn() },
    match: {
      param: {
        id: 1
      }
    }
  };
  const preventDefault = jest.fn();
  const component = mount(<MemoryRouter><Register /></MemoryRouter>)
  it('should render without crashing', () => {
    const { enzymeWrapper } = mount(
      <MemoryRouter>
        <Register {...props} />
      </MemoryRouter>
    );
  });
  it('should render form', () => {
    expect(component.find('Form').length).toBe(1);
    expect(component.find('Form').simulate("submit", { preventDefault }));
    expect(preventDefault).toBeCalled();
  });
});
