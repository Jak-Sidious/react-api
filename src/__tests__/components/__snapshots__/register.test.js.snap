// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register/> should render properly 1`] = `
<div
  className="regBackground"
>
  <_class
    options={Object {}}
  />
  <div
    className="wrapper"
  >
    <Form
      as="form"
      className="form-signin"
      onSubmit={[Function]}
    >
      <h2
        className="form-signin-heading"
      >
        Registration form
      </h2>
      <FormInput
        as={[Function]}
        autoFocus={true}
        className="nameField"
        control={[Function]}
        name="username"
        onChange={[Function]}
        placeholder="Enter your username"
        type="username"
      />
      <FormInput
        as={[Function]}
        autoFocus={true}
        className="emailField"
        control={[Function]}
        name="email"
        onChange={[Function]}
        placeholder="Enter email"
        type="email"
      />
      <FormInput
        as={[Function]}
        autoFocus={true}
        className="passwordField"
        control={[Function]}
        name="password"
        onChange={[Function]}
        placeholder="Enter password"
        type="password"
      />
      <Button
        as="button"
        className="left floated"
        color="blue"
        id="register"
        size="large"
        type="submit"
        value="submit"
      >
        Register
      </Button>
      <Button
        action="/login"
        as={[Function]}
        className="right floated"
        color="green"
        size="large"
        to="/login"
      >
        Proceed to Login
      </Button>
      <br />
    </Form>
  </div>
</div>
`;
